internal/nfs/controller/volume.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2022 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/ceph/ceph-csi/internal/util"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+)
+
+type NFSVolume struct {
+	volumeID  string
+	clusterID string
+	mons      string
+
+	// TODO: drop in favor of a go-ceph connection
+	connected bool
+	cr        *util.Credentials
+}
+
+func NewNFSVolume(volumeID string) (*NFSVolume, error) {
+	// TODO: validate volume.VolumeContext parameters
+	return &NFSVolume{
+		volumeID: volumeID,
+	}, nil
+}
+
+func (nv *NFSVolume) String() string {
+	return nv.volumeID
+}
+
+func (nv *NFSVolume) Connect(cr *util.Credentials) error {
+	nv.cr = cr
+
+	vi := util.CSIIdentifier{}
+
+	err := vi.DecomposeCSIID(nv.volumeID)
+	if err != nil {
+		return fmt.Errorf("error decoding volume ID (%s): %w", nv.volumeID, err)
+	}
+
+	nv.clusterID = vi.ClusterID
+	nv.mons, err = util.Mons(util.CsiConfigFile, vi.ClusterID)
+	if err != nil {
+		return fmt.Errorf("failed to get MONs for cluster (%s): %w", vi.ClusterID, err)
+	}
+
+	nv.connected = true
+
+	return nil
+}
+
+func (nv *NFSVolume) GetExportPath() string {
```

Comment:
add comment for all the exported functions?

Reply:
yes, definitely need to do this!

Reply:
still need to be addressed?